### DISCUSSION TITLE
How are the NaNs in the target variable (streamflow) dealt by the code?

### DISCUSSION THREAD

**Discussion by Meelisha** [neuralhydrology/neuralhydrology#116] [Q&A]
I was wondering how the neuralhydrology code deals with the NaNs in the target variable? I understand [validate_samples](https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/datasetzoo/basedataset.py#L759) ensures all samples in the target are not NaN. However, how does the code work with the target NaNs encountered during the training? Are those samples ignored? Can we interpolate and fill those nan values or backfill or forward fill them? Would filling the empty target samples impact the results and their validity?

**Comment by kratzert** [comment]
Samples, where all targets are NaN are ignored during training. If at least one timestep has a valid target, the sample is included. When we compute the loss, only timesteps with valid labels are included.

Anything else that you want to do (e.g. interpolation/filling) would be something that you need to implement yourself (e.g. in the discharge loading function of the specific dataset class) or outside of NH and simply save a copy of the dataset with the filled values. 

If it would impact the results: Sure. Why wouldn't it?
